src/couchdb/mango-design-document.ts

```diff
@@ -0,0 +1,74 @@
+import { MangoSelector } from 'nano'
+
+interface SortItem {
+  [field: string]: 'asc' | 'desc'
+}
+export type SortSyntax = string | SortItem | Array<string | SortItem>
+
+export interface MangoView {
+  map: {
+    fields: SortItem
+    partial_filter_selector: MangoSelector
+  }
+  reduce: '_count'
+  options: {
+    def: {
+      fields: SortSyntax
+      partial_filter_selector?: MangoSelector
+    }
+  }
+}
+
+export interface MangoDesignDocument {
+  language: 'query'
+  options?: {
+    partitioned?: boolean
+  }
+  views?: { [name: string]: MangoView }
+}
+
+/**
+ * Builds a CouchDB design document that describes a Mango index.
+ */
+export function makeIndexDocument(
```

Comment:
I am curious about the naming of this function; is an index document exclusive to the Mango engine? I'm asking whether we should include Mango in the name.

Reply:
I am open to other names.

Reply:
Picked `makeMangoIndex`.